src/datasource.ts

```diff
@@ -378,12 +378,14 @@ const dynamicSchemaParser = (frames: DataFrame[]): Record<string, AdxColumnSchem
 
 const recordSchemaArray = (name: string, types: AdxSchemaDefinition[], result: AdxColumnSchema[]) => {
   // If a field can have different types (e.g. long and double)
-  // we select the first, assuming they are interchangeable
-  const defaultCslType = types[0];
-  if (types.every((t) => typeof t === 'string' && toPropertyType(t) === QueryEditorPropertyType.Number)) {
-    // If all the types are numbers, the double takes precedence since it has more precission.
-    const cslType = types.find((t) => typeof t === 'string' && (t === 'double' || t === 'real')) || defaultCslType;
-    result.push({ Name: name, CslType: cslType as string, isDynamic: true });
+  // we select double if it exists as it's the one with more precision, otherwise we take the first
+  const defaultCslType = types.find((t) => typeof t === 'string' && (t === 'double' || t === 'real')) || types[0];
+  if (
+    types.length < 2 ||
```

Comment:
I understand that if `cslType` is an array it will have more than one value, but I figured it would be good to catch the single value array case to avoid triggering a warning unnecessarily.

Plus it might come in handy when we make the changes to update the stored schema based on the query filters.

Reply:
I believe that `buildschema` does not return an array if there is only one type but it doesn't hurt to have this check here.

Reply:
also, just a minor styling thing  but now I would order this a bit different. Something like:

```ts
    if (length > 1 && !allNumbers) { console.warn... }
    // previus else, no need to cast defaultCslType as string
    if (typeof defaultCslType === 'object') {
      recordSchema(name, types[0], result);
    } else {
      result.push({ Name: name, CslType: defaultCslType, isDynamic: true });
    }
```

and then you do the rest o